backend/tests/test_chmeme_gen.py

```diff
@@ -0,0 +1,263 @@
+        json.dump(data, f, ensure_ascii=False, indent=4)
+
+
+def get_dataset(data: dict, prompt_path: str, client_endpoint: str, prefix: str, select_numbers: list = None) -> dict:
+    #data_dict=get_tot_data(csv_file_path)
+    data_dict = data
+    if select_numbers is None:
+        random_numbers = get_smaples_random(data_dict, 10)
+    else:
+        random_numbers = select_numbers
+
+    random_numbers.sort()
+    image_list = []
+
+    #img_l=[656,1371,1558,1584,656,908,186,1037,4410,5554,6004,5976,5609,4299]
+    #can be modified
+    img_l = random_numbers
+
+    for num in img_l:
+        img_path = f'{prefix}({num}).jpg'
+        image_list.append(img_path)
+
+    client = OpenAI(
+        # get API KEY from env
+        api_key=api_key,
+        base_url=model_url,
+        #base_url="https://ark.cn-beijing.volces.com/api/v3",
+    )
+    cared_keys = ['sentiment category', 'sentiment degree', 'intention detection', 'offensiveness detection']
+
+    with open(prompt_path, 'r', encoding='utf-8') as file:
+        PROMPT = file.read()
+    ch_dataset = get_ans_chmeme_ch(data_dict, image_list, client, client_endpoint, cared_keys, PROMPT)
+
+    #image_list=['/mnt/afs/niuyazhe/data/meme/data/Cimages/Cimages/Cimages/Image_(272).jpg']
+    json_dataset = {}
+    for num in img_l:
+        img_path = f'{prefix}({num}).jpg'
+        new_img_path = f'{num}'
+        cur_data = {}
+        cur_details = ch_dataset[img_path]
+        text_details = cur_details['description']
+        lines = text_details.strip().split('\n')
+        for line in lines:
+            if line.startswith("(0)"):
+                cur_data["content"] = line[3:].strip()
+
+            elif line.startswith("(1)"):
+                cur_data["emotion"] = line[3:].strip()
+
+            elif line.startswith("(2)"):
+                cur_data["intention"] = line[3:].strip()
+
+            elif line.startswith("(3)"):
+                cur_data["scene"] = line[3:].strip()
+        json_dataset[new_img_path] = cur_data
+
+    return json_dataset
+
+
+#csv_file_path='/mnt/afs/wangqijian/meme/ch_memes/label_C.csv'
+csv_file_path = os.environ.get("csv_file_path")
+#path to the image files
+#image_prefix='/mnt/afs/niuyazhe/data/meme/data/Cimages/Cimages/Cimages/Image_'
+image_prefix = os.environ.get("img_prefix")
+#path to the prompt
+#prompt_path='/mnt/afs/wangqijian/meme/ch_memes/prompt_gen_test_chmemes_ch.txt'
+prompt_path = os.environ.get("prompt_path")
+#path to save the dataset
+#dataset_path_json='/mnt/afs/wangqijian/meme/ch_memes/dataset_4.json'
+dataset_path_json = os.environ.get("data_path_json")
+#new_json_path='/mnt/afs/wangqijian/meme/ch_memes/test.json'
+#need to set API_KEY
+api_key = os.environ.get("ARK_API_KEY")
+#set model_url
+model_url = "https://ark.cn-beijing.volces.com/api/v3"
+#first need to set model_endpoint in website
+model_endpoint = "ep-20250124115350-pwjxt"
+
+
+def main():
+    data = get_tot_data(csv_file_path)
+    processed_numbers = [
```

Comment:
remove processed numbers and related codes, it is just your runtime variables, not for the demo